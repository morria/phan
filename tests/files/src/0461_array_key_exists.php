<?php

/**
 * @param string $class
 * Based on example from https://github.com/phan/phan/issues/1587
 */
function a461($class) {
    $parameters = [];

    if (@array_key_exists("class", $parameters)) {
        $foreign_class = $parameters['class'];
    } else {
        $foreign_class = $class;
    }

    $object = new $foreign_class();
}

a461("stdClass");
