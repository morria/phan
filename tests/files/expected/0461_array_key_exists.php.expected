%s:16 PhanUnusedVariable Unused definition of variable $object
